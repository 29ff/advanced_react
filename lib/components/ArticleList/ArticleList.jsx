import React from 'react';

// import components
import Article from '../Article/Article';

const ArticleList = (props) => {
  const { articles, articleAction } = props;

  return (
    <div>
      {
        Object.keys(articles).map((key) => {
          return <Article
            key={key}
            article={articles[key]}
            articleAction={articleAction}
          />
        })
      }
    </div>
  );
};

export default ArticleList;