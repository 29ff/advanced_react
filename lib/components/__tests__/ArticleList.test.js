import React from 'react';
import ArticleList from '../ArticleList/ArticleList';

import { shallow, configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Article List', () => {
  it('renders correctly', () => {
    const testProps = {
      articles: {
        a: { id: 'a', title: 'a', body: 'a', date: 'a' },
        b: { id: 'b', title: 'b', body: 'b', date: 'b' },
        c: { id: 'c', title: 'c', body: 'c', date: 'c' }
      }
    }
    const wrapper = shallow(
      <ArticleList
        {...testProps}
      />
    );
  })
})
