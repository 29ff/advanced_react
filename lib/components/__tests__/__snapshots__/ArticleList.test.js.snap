// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article List renders correctly 1`] = `
<div>
  <div
    id="article"
  >
    <h3
      id="title-article"
    >
      <strong />
    </h3>
    <div
      id="date-article"
    >
      <strong>
        Invalid date
      </strong>
    </div>
    <div
      id="author-article"
    >
      Author: 
      <a
        href={undefined}
        target="_blank"
      >
        <em>
           
        </em>
      </a>
    </div>
    <div
      id="body-article"
    />
  </div>
  <div
    id="article"
  >
    <h3
      id="title-article"
    >
      <strong />
    </h3>
    <div
      id="date-article"
    >
      <strong>
        Invalid date
      </strong>
    </div>
    <div
      id="author-article"
    >
      Author: 
      <a
        href={undefined}
        target="_blank"
      >
        <em>
           
        </em>
      </a>
    </div>
    <div
      id="body-article"
    />
  </div>
  <div
    id="article"
  >
    <h3
      id="title-article"
    >
      <strong />
    </h3>
    <div
      id="date-article"
    >
      <strong>
        Invalid date
      </strong>
    </div>
    <div
      id="author-article"
    >
      Author: 
      <a
        href={undefined}
        target="_blank"
      >
        <em>
           
        </em>
      </a>
    </div>
    <div
      id="body-article"
    />
  </div>
</div>
`;
